tools/android_custom_build/build_custom_android_package.py

```diff
@@ -168,6 +168,8 @@ def main():
         container_ops_config_file,
         container_build_settings_file,
         "/workspace/shared/output",
+        "-e",
+        'JAVA_OPTS="-Xms16G -Xmx16G"',
```

Comment:
These should probably be earlier in the `docker run` command, otherwise they would be additional arguments passed to `/bin/bash /workspace/scripts/build.sh ...`.

Why are both `-Xms` and `-Xmx` needed?

We have a CI test that regularly [runs this script](https://github.com/microsoft/onnxruntime/blob/8fadc6c913bc30edff2e89756da515b9bd75d256/tools/ci_build/github/azure-pipelines/post-merge-jobs.yml#L429-L436) and it hasn't been failing consistently. Are these settings dependent on the machine? If so, I'd prefer to have a more general way to customize this instead of hardcoding `JAVA_OPTS` here. E.g., this script could provide an additional option to specify docker run environment variable values.

Reply:
Sure, makes sense the flags should occur earlier.

In my case the build crashes if either flag is not present. This probably means the initial allocation and max allocation needs to be at 16GB.

I am not sure if this issue is commonly faced by others, but in my case the crash is reproducible if the flags are not applied. I think providing options to specify docker run env values to the script would be extremely helpful.

Reply:
Leaving the min memory size unset crashes the build with an out of memory error? That sounds very odd. The stack trace you posted is an error out of clang, which shouldn't be sharing memory with the Java process. Does the docker container itself not have enough memory?

Reply:
I am still running a custom Docker image with more memory information logs, to understand the actual behaviour of insufficient memory that causes this issue.

In the meantime, I read https://github.com/microsoft/onnxruntime/pull/19630 and think it is enough to solve https://github.com/microsoft/onnxruntime/issues/19584. It is also a more generic solution as compared to the proposed one. Therefore, feel free to close this PR - I can update more information in the issue conversation.

Reply:
Hi all, after some investigation, can confirm that https://github.com/microsoft/onnxruntime/issues/19584 is an individual machine issue (that machine is faulty and often overheated, which throws random errors at very high temperature (~100 celsius)). We ran the build script on another machine and builds successfully. Can also confirm that the JVM fix is irrelevant to the issue. 

We do face OOM issue sometimes if the Docker container has insufficient memory (as I observe that the build process might need up to 16+ GB of memory), and can be solved by adding Docker env variables e.g. `-m 32g`. Which means https://github.com/microsoft/onnxruntime/pull/19630 should be a helpful fix. 

I will proceed to close this PR as it is no longer relevant. Thank you again for you help.

Reply:
> We do face OOM issue sometimes if the Docker container has insufficient memory (as I observe that the build process might need up to 16+ GB of memory), and can be solved by adding Docker env variables e.g. -m 32g. Which means https://github.com/microsoft/onnxruntime/pull/19630 should be a helpful fix.

FYI, PR #19630 only allows one to specify additional `--env` options to docker run and not other options like `-m`. Maybe we don't need that capability for now. Can reconsider adding it later if we have a reason to.